predicators/approaches/bridge_policy_approach.py

```diff
@@ -486,13 +569,183 @@ def learn_from_interaction_results(
+            if sesame_task_planner.endswith("-costs"):
+                use_costs = True
+                sesame_task_planner = sesame_task_planner[:-len("-costs")]
+            else:
+                use_costs = False
+
+            if sesame_task_planner == "fdopt":
+                alias_flag = "--alias seq-opt-lmcut"
+            elif sesame_task_planner == "fdsat":
+                alias_flag = "--alias lama-first"
+            else:
+                raise ValueError("Unrecognized sesame_task_planner: "
+                                 f"{CFG.sesame_task_planner}")
+
+            sas_file = generate_sas_file_for_fd(task, nsrts, preds, types,
+                                                timeout, timeout_cmd,
+                                                alias_flag, exec_str,
+                                                list(objects), init_atoms)
+
+            if use_costs:
+                assert ground_op_costs is not None
+                assert all(c >= 0 for c in ground_op_costs.values())
+                _update_sas_file_with_costs(
+                    sas_file,
+                    ground_op_costs,
+                    default_ground_op_cost=default_cost,
+                    cost_precision=cost_precision)
+
+            plan, atoms_seq, metrics = fd_plan_from_sas_file(
+                sas_file, timeout_cmd, timeout, exec_str,
+                alias_flag, start_time, list(objects), init_atoms, nsrts,
+                float(max_horizon))
+        else:
+            raise ValueError("Unrecognized sesame_task_planner: "
+                             f"{CFG.sesame_task_planner}")
+
+        necessary_atoms_seq = utils.compute_necessary_atoms_seq(
+            plan, atoms_seq, goal)
+
+        return plan, necessary_atoms_seq, metrics
+
+    def _get_option_policy_by_planning( #pylint: disable=dangerous-default-value
+            self, task: Task,
+            timeout: float, init: Set[GroundAtom] = set()) \
+                -> Callable[[State], _Option]:
+        """Raises an OptionExecutionFailure with the last_failed_option in its
+        info dict in the case where execution fails."""
+
+        # Ensure random over successive calls.
+        self._num_calls += 1
+        seed = self._seed + self._num_calls
+        nsrts = self._get_current_nsrts()
+        preds = self._get_current_predicates()
+
+        nsrt_plan, atoms_seq, _ = self._run_task_plan(task, nsrts, preds,
+                                                      timeout, seed, init)
+        return utils.nsrt_plan_to_greedy_option_policy(
+            nsrt_plan,
+            goal=task.goal,
+            rng=self._rng,
+            necessary_atoms_seq=atoms_seq)
+
+    def _run_task_plan( #pylint: disable=dangerous-default-value
+        self, task: Task, nsrts: Set[NSRT],
+        preds: Set[Predicate], timeout: float, seed: int, \
+            init: Set[GroundAtom] = set(), **kwargs: Any
+    ) -> Tuple[List[_GroundNSRT], List[Set[GroundAtom]], Metrics]:
```

Comment:
okay - i'm not fully sure why you had to override this function here? Why not use the function from `planning.py`?

Reply:
I needed to override it to include the new init variable so we can start the plan from any state.